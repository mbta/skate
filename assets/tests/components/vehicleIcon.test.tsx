import React from "react"
import renderer from "react-test-renderer"
import VehicleIcon, {
  Orientation,
  Size,
  VehicleIconSvgNode,
} from "../../src/components/vehicleIcon"

test("renders in all directions and sizes", () => {
  const tree = renderer
    .create(
      <>
        <VehicleIcon
          label="label"
          size={Size.Small}
          orientation={Orientation.Up}
        />
        <VehicleIcon
          label="label"
          size={Size.Small}
          orientation={Orientation.Right}
        />
        <VehicleIcon
          label="label"
          size={Size.Small}
          orientation={Orientation.Down}
        />
        <VehicleIcon
          label="label"
          size={Size.Small}
          orientation={Orientation.Left}
        />
        <VehicleIcon
          label="label"
          size={Size.Medium}
          orientation={Orientation.Up}
        />
        <VehicleIcon
          label="label"
          size={Size.Medium}
          orientation={Orientation.Right}
        />
        <VehicleIcon
          label="label"
          size={Size.Medium}
          orientation={Orientation.Down}
        />
        <VehicleIcon
          label="label"
          size={Size.Medium}
          orientation={Orientation.Left}
        />
        <VehicleIcon
          label="label"
          size={Size.Large}
          orientation={Orientation.Up}
        />
        <VehicleIcon
          label="label"
          size={Size.Large}
          orientation={Orientation.Right}
        />
        <VehicleIcon
          label="label"
          size={Size.Large}
          orientation={Orientation.Down}
        />
        <VehicleIcon
          label="label"
          size={Size.Large}
          orientation={Orientation.Left}
        />
      </>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})

test("renders with variants and labels", () => {
  const tree = renderer
    .create(
      <>
        <VehicleIcon
          size={Size.Small}
          orientation={Orientation.Up}
          label="0617"
          variant="X"
        />
        <VehicleIcon
          size={Size.Small}
          orientation={Orientation.Right}
          label="0617"
          variant="X"
        />
        <VehicleIcon
          size={Size.Small}
          orientation={Orientation.Down}
          label="0617"
          variant="X"
        />
        <VehicleIcon
          size={Size.Small}
          orientation={Orientation.Left}
          label="0617"
          variant="X"
        />
        <VehicleIcon
          size={Size.Medium}
          orientation={Orientation.Up}
          label="0617"
          variant="X"
        />
        <VehicleIcon
          size={Size.Medium}
          orientation={Orientation.Right}
          label="0617"
          variant="X"
        />
        <VehicleIcon
          size={Size.Medium}
          orientation={Orientation.Down}
          label="0617"
          variant="X"
        />
        <VehicleIcon
          size={Size.Medium}
          orientation={Orientation.Left}
          label="0617"
          variant="X"
        />
        <VehicleIcon
          size={Size.Large}
          orientation={Orientation.Up}
          label="0617"
          variant="X"
        />
        <VehicleIcon
          size={Size.Large}
          orientation={Orientation.Right}
          label="0617"
          variant="X"
        />
        <VehicleIcon
          size={Size.Large}
          orientation={Orientation.Down}
          label="0617"
          variant="X"
        />
        <VehicleIcon
          size={Size.Large}
          orientation={Orientation.Left}
          label="0617"
          variant="X"
        />
      </>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})

test("renders an unwrapped svg node", () => {
  const tree = renderer
    .create(
      <svg>
        <VehicleIconSvgNode
          size={Size.Medium}
          orientation={Orientation.Down}
          label="0617"
        />
      </svg>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
