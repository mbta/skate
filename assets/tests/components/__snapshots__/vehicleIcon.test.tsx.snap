// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an unwrapped svg node 1`] = `
<svg>
  <g
    className="m-vehicle-icon m-vehicle-icon--medium "
  >
    <rect
      className="m-vehicle-icon__label-background"
      height={11}
      rx={5.5}
      ry={5.5}
      width={26}
      x={-13}
      y={-22.5}
    />
    <text
      className="m-vehicle-icon__label"
      dominantBaseline="central"
      textAnchor="middle"
      x="0"
      y={-17}
    >
      label
    </text>
    <path
      className="m-vehicle-icon__triangle"
      d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
      transform="scale(0.625) rotate(180) translate(-24,-22)"
    />
  </g>
</svg>
`;

exports[`renders in all directions and sizes 1`] = `
Array [
  <svg
    style={
      Object {
        "height": 13.68,
        "width": 16.72,
      }
    }
    viewBox="-8.36 -6.08 16.72 13.68"
  >
    <g
      className="m-vehicle-icon m-vehicle-icon--small "
    >
      <path
        className="m-vehicle-icon__triangle"
        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
        transform="scale(0.38) rotate(0) translate(-24,-22)"
      />
    </g>
  </svg>,
  <svg
    style={
      Object {
        "height": 16.72,
        "width": 13.68,
      }
    }
    viewBox="-7.6 -8.36 13.68 16.72"
  >
    <g
      className="m-vehicle-icon m-vehicle-icon--small "
    >
      <path
        className="m-vehicle-icon__triangle"
        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
        transform="scale(0.38) rotate(90) translate(-24,-22)"
      />
    </g>
  </svg>,
  <svg
    style={
      Object {
        "height": 13.68,
        "width": 16.72,
      }
    }
    viewBox="-8.36 -7.6 16.72 13.68"
  >
    <g
      className="m-vehicle-icon m-vehicle-icon--small "
    >
      <path
        className="m-vehicle-icon__triangle"
        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
        transform="scale(0.38) rotate(180) translate(-24,-22)"
      />
    </g>
  </svg>,
  <svg
    style={
      Object {
        "height": 16.72,
        "width": 13.68,
      }
    }
    viewBox="-6.08 -8.36 13.68 16.72"
  >
    <g
      className="m-vehicle-icon m-vehicle-icon--small "
    >
      <path
        className="m-vehicle-icon__triangle"
        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
        transform="scale(0.38) rotate(270) translate(-24,-22)"
      />
    </g>
  </svg>,
  <svg
    style={
      Object {
        "height": 22.5,
        "width": 27.5,
      }
    }
    viewBox="-13.75 -10 27.5 22.5"
  >
    <g
      className="m-vehicle-icon m-vehicle-icon--medium "
    >
      <path
        className="m-vehicle-icon__triangle"
        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
        transform="scale(0.625) rotate(0) translate(-24,-22)"
      />
    </g>
  </svg>,
  <svg
    style={
      Object {
        "height": 27.5,
        "width": 22.5,
      }
    }
    viewBox="-12.5 -13.75 22.5 27.5"
  >
    <g
      className="m-vehicle-icon m-vehicle-icon--medium "
    >
      <path
        className="m-vehicle-icon__triangle"
        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
        transform="scale(0.625) rotate(90) translate(-24,-22)"
      />
    </g>
  </svg>,
  <svg
    style={
      Object {
        "height": 22.5,
        "width": 27.5,
      }
    }
    viewBox="-13.75 -12.5 27.5 22.5"
  >
    <g
      className="m-vehicle-icon m-vehicle-icon--medium "
    >
      <path
        className="m-vehicle-icon__triangle"
        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
        transform="scale(0.625) rotate(180) translate(-24,-22)"
      />
    </g>
  </svg>,
  <svg
    style={
      Object {
        "height": 27.5,
        "width": 22.5,
      }
    }
    viewBox="-10 -13.75 22.5 27.5"
  >
    <g
      className="m-vehicle-icon m-vehicle-icon--medium "
    >
      <path
        className="m-vehicle-icon__triangle"
        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
        transform="scale(0.625) rotate(270) translate(-24,-22)"
      />
    </g>
  </svg>,
  <svg
    style={
      Object {
        "height": 36,
        "width": 44,
      }
    }
    viewBox="-22 -16 44 36"
  >
    <g
      className="m-vehicle-icon m-vehicle-icon--large "
    >
      <path
        className="m-vehicle-icon__triangle"
        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
        transform="scale(1) rotate(0) translate(-24,-22)"
      />
    </g>
  </svg>,
  <svg
    style={
      Object {
        "height": 44,
        "width": 36,
      }
    }
    viewBox="-20 -22 36 44"
  >
    <g
      className="m-vehicle-icon m-vehicle-icon--large "
    >
      <path
        className="m-vehicle-icon__triangle"
        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
        transform="scale(1) rotate(90) translate(-24,-22)"
      />
    </g>
  </svg>,
  <svg
    style={
      Object {
        "height": 36,
        "width": 44,
      }
    }
    viewBox="-22 -20 44 36"
  >
    <g
      className="m-vehicle-icon m-vehicle-icon--large "
    >
      <path
        className="m-vehicle-icon__triangle"
        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
        transform="scale(1) rotate(180) translate(-24,-22)"
      />
    </g>
  </svg>,
  <svg
    style={
      Object {
        "height": 44,
        "width": 36,
      }
    }
    viewBox="-16 -22 36 44"
  >
    <g
      className="m-vehicle-icon m-vehicle-icon--large "
    >
      <path
        className="m-vehicle-icon__triangle"
        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
        transform="scale(1) rotate(270) translate(-24,-22)"
      />
    </g>
  </svg>,
]
`;

exports[`renders with all statuses 1`] = `
Array [
  <svg
    style={
      Object {
        "height": 22.5,
        "width": 27.5,
      }
    }
    viewBox="-13.75 -10 27.5 22.5"
  >
    <g
      className="m-vehicle-icon m-vehicle-icon--medium on-time"
    >
      <path
        className="m-vehicle-icon__triangle"
        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
        transform="scale(0.625) rotate(0) translate(-24,-22)"
      />
    </g>
  </svg>,
  <svg
    style={
      Object {
        "height": 22.5,
        "width": 27.5,
      }
    }
    viewBox="-13.75 -10 27.5 22.5"
  >
    <g
      className="m-vehicle-icon m-vehicle-icon--medium early"
    >
      <path
        className="m-vehicle-icon__triangle"
        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
        transform="scale(0.625) rotate(0) translate(-24,-22)"
      />
    </g>
  </svg>,
  <svg
    style={
      Object {
        "height": 22.5,
        "width": 27.5,
      }
    }
    viewBox="-13.75 -10 27.5 22.5"
  >
    <g
      className="m-vehicle-icon m-vehicle-icon--medium late"
    >
      <path
        className="m-vehicle-icon__triangle"
        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
        transform="scale(0.625) rotate(0) translate(-24,-22)"
      />
    </g>
  </svg>,
  <svg
    style={
      Object {
        "height": 22.5,
        "width": 27.5,
      }
    }
    viewBox="-13.75 -10 27.5 22.5"
  >
    <g
      className="m-vehicle-icon m-vehicle-icon--medium off-course"
    >
      <path
        className="m-vehicle-icon__triangle"
        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
        transform="scale(0.625) rotate(0) translate(-24,-22)"
      />
    </g>
  </svg>,
  <svg
    style={
      Object {
        "height": 22.5,
        "width": 27.5,
      }
    }
    viewBox="-13.75 -10 27.5 22.5"
  >
    <g
      className="m-vehicle-icon m-vehicle-icon--medium "
    >
      <path
        className="m-vehicle-icon__triangle"
        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
        transform="scale(0.625) rotate(0) translate(-24,-22)"
      />
    </g>
  </svg>,
  <svg
    style={
      Object {
        "height": 22.5,
        "width": 27.5,
      }
    }
    viewBox="-13.75 -10 27.5 22.5"
  >
    <g
      className="m-vehicle-icon m-vehicle-icon--medium "
    >
      <path
        className="m-vehicle-icon__triangle"
        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
        transform="scale(0.625) rotate(0) translate(-24,-22)"
      />
    </g>
  </svg>,
]
`;

exports[`renders with variants and labels 1`] = `
Array [
  <svg
    style={
      Object {
        "height": 24.68,
        "width": 26,
      }
    }
    viewBox="-13 -6.08 26 24.68"
  >
    <g
      className="m-vehicle-icon m-vehicle-icon--small "
    >
      <rect
        className="m-vehicle-icon__label-background"
        height={11}
        rx={5.5}
        ry={5.5}
        width={26}
        x={-13}
        y={6.6}
      />
      <text
        className="m-vehicle-icon__label"
        dominantBaseline="central"
        textAnchor="middle"
        x="0"
        y={12.1}
      >
        0617
      </text>
      <path
        className="m-vehicle-icon__triangle"
        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
        transform="scale(0.38) rotate(0) translate(-24,-22)"
      />
      <text
        className="m-vehicle-icon__variant"
        dominantBaseline="alphabetic"
        textAnchor="middle"
        x={0}
        y={5.6}
      >
        X
      </text>
    </g>
  </svg>,
  <svg
    style={
      Object {
        "height": 27.72,
        "width": 26,
      }
    }
    viewBox="-13 -8.36 26 27.72"
  >
    <g
      className="m-vehicle-icon m-vehicle-icon--small "
    >
      <rect
        className="m-vehicle-icon__label-background"
        height={11}
        rx={5.5}
        ry={5.5}
        width={26}
        x={-13}
        y={7.359999999999999}
      />
      <text
        className="m-vehicle-icon__label"
        dominantBaseline="central"
        textAnchor="middle"
        x="0"
        y={12.86}
      >
        0617
      </text>
      <path
        className="m-vehicle-icon__triangle"
        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
        transform="scale(0.38) rotate(90) translate(-24,-22)"
      />
      <text
        className="m-vehicle-icon__variant"
        dominantBaseline="central"
        textAnchor="start"
        x={-5.6}
        y={0}
      >
        X
      </text>
    </g>
  </svg>,
  <svg
    style={
      Object {
        "height": 24.68,
        "width": 26,
      }
    }
    viewBox="-13 -18.6 26 24.68"
  >
    <g
      className="m-vehicle-icon m-vehicle-icon--small "
    >
      <rect
        className="m-vehicle-icon__label-background"
        height={11}
        rx={5.5}
        ry={5.5}
        width={26}
        x={-13}
        y={-17.6}
      />
      <text
        className="m-vehicle-icon__label"
        dominantBaseline="central"
        textAnchor="middle"
        x="0"
        y={-12.100000000000001}
      >
        0617
      </text>
      <path
        className="m-vehicle-icon__triangle"
        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
        transform="scale(0.38) rotate(180) translate(-24,-22)"
      />
      <text
        className="m-vehicle-icon__variant"
        dominantBaseline="hanging"
        textAnchor="middle"
        x={0}
        y={-5.6}
      >
        X
      </text>
    </g>
  </svg>,
  <svg
    style={
      Object {
        "height": 27.72,
        "width": 26,
      }
    }
    viewBox="-13 -8.36 26 27.72"
  >
    <g
      className="m-vehicle-icon m-vehicle-icon--small "
    >
      <rect
        className="m-vehicle-icon__label-background"
        height={11}
        rx={5.5}
        ry={5.5}
        width={26}
        x={-13}
        y={7.359999999999999}
      />
      <text
        className="m-vehicle-icon__label"
        dominantBaseline="central"
        textAnchor="middle"
        x="0"
        y={12.86}
      >
        0617
      </text>
      <path
        className="m-vehicle-icon__triangle"
        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
        transform="scale(0.38) rotate(270) translate(-24,-22)"
      />
      <text
        className="m-vehicle-icon__variant"
        dominantBaseline="central"
        textAnchor="end"
        x={5.6}
        y={0}
      >
        X
      </text>
    </g>
  </svg>,
  <svg
    style={
      Object {
        "height": 33.5,
        "width": 27.5,
      }
    }
    viewBox="-13.75 -10 27.5 33.5"
  >
    <g
      className="m-vehicle-icon m-vehicle-icon--medium "
    >
      <rect
        className="m-vehicle-icon__label-background"
        height={11}
        rx={5.5}
        ry={5.5}
        width={26}
        x={-13}
        y={11.5}
      />
      <text
        className="m-vehicle-icon__label"
        dominantBaseline="central"
        textAnchor="middle"
        x="0"
        y={17}
      >
        0617
      </text>
      <path
        className="m-vehicle-icon__triangle"
        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
        transform="scale(0.625) rotate(0) translate(-24,-22)"
      />
      <text
        className="m-vehicle-icon__variant"
        dominantBaseline="alphabetic"
        textAnchor="middle"
        x={0}
        y={8.5}
      >
        X
      </text>
    </g>
  </svg>,
  <svg
    style={
      Object {
        "height": 38.5,
        "width": 26,
      }
    }
    viewBox="-13 -13.75 26 38.5"
  >
    <g
      className="m-vehicle-icon m-vehicle-icon--medium "
    >
      <rect
        className="m-vehicle-icon__label-background"
        height={11}
        rx={5.5}
        ry={5.5}
        width={26}
        x={-13}
        y={12.75}
      />
      <text
        className="m-vehicle-icon__label"
        dominantBaseline="central"
        textAnchor="middle"
        x="0"
        y={18.25}
      >
        0617
      </text>
      <path
        className="m-vehicle-icon__triangle"
        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
        transform="scale(0.625) rotate(90) translate(-24,-22)"
      />
      <text
        className="m-vehicle-icon__variant"
        dominantBaseline="central"
        textAnchor="start"
        x={-8.5}
        y={0}
      >
        X
      </text>
    </g>
  </svg>,
  <svg
    style={
      Object {
        "height": 33.5,
        "width": 27.5,
      }
    }
    viewBox="-13.75 -23.5 27.5 33.5"
  >
    <g
      className="m-vehicle-icon m-vehicle-icon--medium "
    >
      <rect
        className="m-vehicle-icon__label-background"
        height={11}
        rx={5.5}
        ry={5.5}
        width={26}
        x={-13}
        y={-22.5}
      />
      <text
        className="m-vehicle-icon__label"
        dominantBaseline="central"
        textAnchor="middle"
        x="0"
        y={-17}
      >
        0617
      </text>
      <path
        className="m-vehicle-icon__triangle"
        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
        transform="scale(0.625) rotate(180) translate(-24,-22)"
      />
      <text
        className="m-vehicle-icon__variant"
        dominantBaseline="hanging"
        textAnchor="middle"
        x={0}
        y={-8.5}
      >
        X
      </text>
    </g>
  </svg>,
  <svg
    style={
      Object {
        "height": 38.5,
        "width": 26,
      }
    }
    viewBox="-13 -13.75 26 38.5"
  >
    <g
      className="m-vehicle-icon m-vehicle-icon--medium "
    >
      <rect
        className="m-vehicle-icon__label-background"
        height={11}
        rx={5.5}
        ry={5.5}
        width={26}
        x={-13}
        y={12.75}
      />
      <text
        className="m-vehicle-icon__label"
        dominantBaseline="central"
        textAnchor="middle"
        x="0"
        y={18.25}
      >
        0617
      </text>
      <path
        className="m-vehicle-icon__triangle"
        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
        transform="scale(0.625) rotate(270) translate(-24,-22)"
      />
      <text
        className="m-vehicle-icon__variant"
        dominantBaseline="central"
        textAnchor="end"
        x={8.5}
        y={0}
      >
        X
      </text>
    </g>
  </svg>,
  <svg
    style={
      Object {
        "height": 60,
        "width": 64,
      }
    }
    viewBox="-32 -16 64 60"
  >
    <g
      className="m-vehicle-icon m-vehicle-icon--large "
    >
      <rect
        className="m-vehicle-icon__label-background"
        height={24}
        rx={12}
        ry={12}
        width={64}
        x={-32}
        y={19}
      />
      <text
        className="m-vehicle-icon__label"
        dominantBaseline="central"
        textAnchor="middle"
        x="0"
        y={31}
      >
        0617
      </text>
      <path
        className="m-vehicle-icon__triangle"
        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
        transform="scale(1) rotate(0) translate(-24,-22)"
      />
      <text
        className="m-vehicle-icon__variant"
        dominantBaseline="alphabetic"
        textAnchor="middle"
        x={0}
        y={14}
      >
        X
      </text>
    </g>
  </svg>,
  <svg
    style={
      Object {
        "height": 68,
        "width": 64,
      }
    }
    viewBox="-32 -22 64 68"
  >
    <g
      className="m-vehicle-icon m-vehicle-icon--large "
    >
      <rect
        className="m-vehicle-icon__label-background"
        height={24}
        rx={12}
        ry={12}
        width={64}
        x={-32}
        y={21}
      />
      <text
        className="m-vehicle-icon__label"
        dominantBaseline="central"
        textAnchor="middle"
        x="0"
        y={33}
      >
        0617
      </text>
      <path
        className="m-vehicle-icon__triangle"
        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
        transform="scale(1) rotate(90) translate(-24,-22)"
      />
      <text
        className="m-vehicle-icon__variant"
        dominantBaseline="central"
        textAnchor="start"
        x={-14}
        y={0}
      >
        X
      </text>
    </g>
  </svg>,
  <svg
    style={
      Object {
        "height": 60,
        "width": 64,
      }
    }
    viewBox="-32 -44 64 60"
  >
    <g
      className="m-vehicle-icon m-vehicle-icon--large "
    >
      <rect
        className="m-vehicle-icon__label-background"
        height={24}
        rx={12}
        ry={12}
        width={64}
        x={-32}
        y={-43}
      />
      <text
        className="m-vehicle-icon__label"
        dominantBaseline="central"
        textAnchor="middle"
        x="0"
        y={-31}
      >
        0617
      </text>
      <path
        className="m-vehicle-icon__triangle"
        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
        transform="scale(1) rotate(180) translate(-24,-22)"
      />
      <text
        className="m-vehicle-icon__variant"
        dominantBaseline="hanging"
        textAnchor="middle"
        x={0}
        y={-14}
      >
        X
      </text>
    </g>
  </svg>,
  <svg
    style={
      Object {
        "height": 68,
        "width": 64,
      }
    }
    viewBox="-32 -22 64 68"
  >
    <g
      className="m-vehicle-icon m-vehicle-icon--large "
    >
      <rect
        className="m-vehicle-icon__label-background"
        height={24}
        rx={12}
        ry={12}
        width={64}
        x={-32}
        y={21}
      />
      <text
        className="m-vehicle-icon__label"
        dominantBaseline="central"
        textAnchor="middle"
        x="0"
        y={33}
      >
        0617
      </text>
      <path
        className="m-vehicle-icon__triangle"
        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
        transform="scale(1) rotate(270) translate(-24,-22)"
      />
      <text
        className="m-vehicle-icon__variant"
        dominantBaseline="central"
        textAnchor="end"
        x={14}
        y={0}
      >
        X
      </text>
    </g>
  </svg>,
]
`;
